### ISSUE TITLE
How to train on custom dataset?

### ISSUE THREAD

**Issue by nisssal** [Xilinx/QNN-MO-PYNQ#3]
Hi,

Could you please give us directions to train this to our own data set? 
Or is it possible to train this network on original yolo implementation and covert the .weights files to the required format? Any script?

**Comment by giuliogamba** [comment]
Hi, 

I guess you are talking about Darknet training. First you will need to instrument Darknet to perform training on reduced precision weights and activations. The modified version used for this repo has not been open-sourced yet. Once the training on your dataset achieves the desired accuracy, you can export the .weights file into the .bin used in this example. The script to performe that will be released in the near future.

**Comment by nisssal** [comment]
Thank you. The .weights conversion script will be really helpful.

**Comment by riple** [comment]
What is the DL framework used in the training process for DoReFaNet? TensorFlow, Theano or MXNet?

**Comment by giuliogamba** [comment]
[Tensorpack](https://github.com/ppwwyyxx/tensorpack) on top of TensorFlow has been used for DorefaNet

**Comment by riple** [comment]
If you use different frameworks for the 2 demo networks, you must have different scripts to covert the weight parameters to a unified form for the FPGA. Is this right?
Will you share the training scripts and converting scripts in the future?

**Comment by giuliogamba** [comment]
Yes, we have planned to release those scripts

**Comment by blgpb** [comment]
Excuse me, when will it be released?

**Comment by Ashsur** [comment]
Excuse me, when will it be released?

**Comment by Kr0n0** [comment]
Hi. Any plans for the scripts release dates? Thx.

**Comment by rafferino** [comment]
Hello, is the script still in progress or has the plan to release it to open-source changed?

**Comment by ussamazahid96** [comment]
For quantized **dorefanet** training in **tensorpack** please have a look at [this](https://github.com/mohdumar644/QNN-MO-PYNQ) repo.

For quantized **dorefanet** training in **pytorch** please have a look at [this](https://github.com/mohdumar644/DoReFaNet-PyTorch) repo.

For quantized **TinyYolo** training (**pytorch**) please have a look at [this](https://github.com/mohdumar644/TinyYOLO-BNN)

Thanks!

**Comment by mohitajais** [comment]
Hello, 

I want to implement quantized tiny yolo on FPGA. I found this link useful -https://github.com/mohdumar644/TinyYOLO-BNN. I have tried to implement quantized Tiny Yolo training (PyTorch) from this link. My question is how the quantization is working in tiny yolo layers. I am getting very low detection accuracy on the PASCAL VOC test set using this quantized model. 

Please Help

> For quantized **dorefanet** training in **tensorpack** please have a look at [this](https://github.com/mohdumar644/QNN-MO-PYNQ) repo.
> 
> For quantized **dorefanet** training in **pytorch** please have a look at [this](https://github.com/mohdumar644/DoReFaNet-PyTorch) repo.
> 
> For quantized **TinyYolo** training (**pytorch**) please have a look at [this](https://github.com/mohdumar644/TinyYOLO-BNN)
> 
> Thanks!
